src/openzaak/utils/migrations.py

```diff
@@ -0,0 +1,103 @@
+# SPDX-License-Identifier: EUPL-1.2
+# Copyright (C) 2022 Dimpact
+import logging
+import re
+from urllib.parse import urlsplit, urlunsplit
+
+from django.db.models.functions import Length
+
+from zgw_consumers.constants import APITypes
+
+logger = logging.getLogger(__name__)
+
+
+def get_service(model, url: str):
+    """
+    copy-paste from zgw_consumers.Service.get_service method, which can't be used
+    during migrations
+    """
+    split_url = urlsplit(url)
+    scheme_and_domain = urlunsplit(split_url[:2] + ("", "", ""))
+
+    candidates = (
+        model.objects.filter(api_root__startswith=scheme_and_domain)
+        .annotate(api_root_length=Length("api_root"))
+        .order_by("-api_root_length")
+    )
+
+    # select the one matching
+    for candidate in candidates.iterator():
+        if url.startswith(candidate.api_root):
+            return candidate
+
+    return None
+
+
+def calculate_api_root(url: str) -> str:
+    """
+    guess apI_root based on following logic:
+    * if url has /api/v{0-9}: api_root = prefix + that part
+    * if url has "/" - include prefix + that part
+    * otherwise include the whole string
+    """
+    patterns = [
+        r"(.+/api/v\d+/).+",
+        r"(\w+://.+\.\w+/).*",
```

Comment:
```suggestion
        r"(https?://.+\.\w+/).*",
```
any reason why we'd support anything that's not http(s)?